app/services/permissions/order.rb

```diff
@@ -2,24 +2,26 @@
 
 module Permissions
   class Order
-    def initialize(user)
+    def initialize(user, params = nil)
       @user = user
       @permissions = OpenFoodNetwork::Permissions.new(@user)
+      @params = params
     end
 
     # Find orders that the user can see
     def visible_orders
-      Spree::Order.
+      orders = Spree::Order.
         with_line_items_variants_and_products_outer.
         where(visible_orders_where_values)
+      orders = orders.complete.not_state(:canceled).search(@params[:q]).result if search_orders?
+      orders
     end
 
     # Any orders that the user can edit
     def editable_orders
-      Spree::Order.where(
-        managed_orders_where_values.
-          or(coordinated_orders_where_values)
-      )
+      orders = Spree::Order.where(managed_orders_where_values.or(coordinated_orders_where_values))
+      orders = orders.complete.not_state(:canceled).search(@params[:q]).result if search_orders?
```

Comment:
Is this change correct? Should the reports only list complete not cancelled orders?

Reply:
did you verify the SQL produced? Does it look good as part of the massive query?

Reply:
if this is correct, I think it needs a spec.

Reply:
> did you verify the SQL produced? Does it look good as part of the massive query? 

I checked the actual output of the queries with real data, using rack-mini-profiler both before and after.

Reply:
ok, great. that means you validated the replacement of the merge with the names scope as well.

Reply:
> Is this change correct? Should the reports only list complete not cancelled orders?

All the reports this PR affects use `Reports::LineItems` to get their line items, which uses this method (unchanged):
https://github.com/openfoodfoundation/openfoodnetwork/blob/981778bc98ce43dde6ff57da8272b997038745be/lib/open_food_network/reports/line_items.rb#L48-L50

Reply:
> if this is correct, I think it needs a spec

Specs added :+1: